### PR TITLE
[FIX] stock: display the customer address instead of the delivery address

### PR DESCRIPTION
**Pull request by Jinjiu96**
Reproduction:
1. Install Inventory and Sales, enable customer address in the setting
of Sale
2. Create a quotation, choose a customer which has different contact
address and delivery address, add a storable product.
3. Confirm the order and click the delivery in the status bar
4. Click print->delivery slip, the Customer Address and Delivery Address
are the same

Reason: The Customer Address isn’t correct in the template

Fix: replace the partner setting in the customer address part as either
the partner or its parent partner_id, e.g. the commercial_partner_id

opw-2851158

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Jinjiu96** [comment]
@AntoineVDV Could you have a look at this fix? Thanks!

**Comment by Jinjiu96** [comment]
Hello @amoyaux , thanks a bunch for the input! I tested it again and indeed it also needed to be fixed without the sale model. However, personally I think it can be a big fix creating the computed field. I made another fix by checking if the partner has a parent_id, in which case the current partner is the sub_partner created for the delivery address. If there's a parent_id, we use the parent partner for the customer address, if there's no parent_id, we keep using the partner. From my test, it works well for the delivery slip comes from a SO and for those I create directly in the inventory app. I attached a few pictures for the result. Could you have a look and let me know how do you like the new fix? Thanks a bunch! :smile:

**Comment by Jinjiu96** [comment]
@amoyaux up :pray:

File: addons/stock/report/report_deliveryslip.xml [hunk 1]
```diff
@@ -33,9 +33,10 @@
                                 <span><strong>Customer Address:</strong></span>
                             </div>
                             <div t-if="partner" name="partner_header">
-                                <div t-field="partner.self"
+                                <t t-set="customer_addr_partner" t-value="partner.parent_id or partner"/>
```
Comment (DjamelTouati) [inline addons/stock/report/report_deliveryslip.xml:36]:
```suggestion
                                <div t-field="partner.commercial_partner_id"
```

File: addons/stock/report/report_deliveryslip.xml [hunk 2]
```diff
@@ -33,9 +33,10 @@
...
+                                <div t-field="customer_addr_partner.self"
                                     t-options='{"widget": "contact", "fields": ["address", "name", "phone"], "no_marker": True, "phone_icons": True}'/>
-                                <p t-if="partner.sudo().vat"><t t-esc="o.company_id.country_id.vat_label or 'Tax ID'"/>: <span t-field="partner.sudo().vat"/></p>
+                                <p t-if="customer_addr_partner.sudo().vat"><t t-esc="o.company_id.country_id.vat_label or 'Tax ID'"/>: <span t-field="customer_addr_partner.sudo().vat"/></p>
```
Comment (DjamelTouati) [inline addons/stock/report/report_deliveryslip.xml:39]:
```suggestion
<p t-if="partner.sudo().commercial_partner_id.vat"><t t-esc="o.company_id.country_id.vat_label or 'Tax ID'"/>: <span t-field="partner.sudo().commercial_partner_id.vat"/></p>
```

**Review by DjamelTouati — CHANGES_REQUESTED** [review]
Hi @Jinjiu96 
Thanks for the fix, I think you could simply do this

Could you also update your commit title please for something like:
[FIX] stock: display the customer address instead of the delivery address

And add these two steps:
- Enable the `customer address` option in the sales settings
- Add a storable product in the quotation

**Review by adwid — APPROVED** [review]
When generating the values of the procurement, we use the shipping address as value of `partner_id`:
https://github.com/odoo/odoo/blob/e11dfa341baf82fedee753502aef5ef2972f75cb/addons/sale_stock/models/sale_order.py#L493
(This value will be defined on the related SM and picking)
So, using `partner_id` as "Customer Address" is incorrect, we should rather use `partner_id.commercial_partner_id`, as suggested by  the fix

robodoo r+

**Comment by Jinjiu96** [comment]
> When generating the values of the procurement, we use the shipping address as value of `partner_id`:
> 
> https://github.com/odoo/odoo/blob/e11dfa341baf82fedee753502aef5ef2972f75cb/addons/sale_stock/models/sale_order.py#L493
> 
> 
> (This value will be defined on the related SM and picking)
> So, using `partner_id` as "Customer Address" is incorrect, we should rather use `partner_id.commercial_partner_id`, as suggested by the fix
> robodoo r+

Thanks for checking! :pray:

**Pull request closed without merging**